sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_obj_impl.hpp

```diff
@@ -745,18 +765,50 @@ ESIMD_INLINE EnableIfAccessor<AccessorT, accessor_mode_cap::can_read,
                               sycl::access::target::global_buffer, void>
 simd_obj_impl<T, N, T1, SFINAE>::copy_from(AccessorT acc, uint32_t offset,
                                            Flags) SYCL_ESIMD_FUNCTION {
-  *this = block_load<T, N, AccessorT, Flags>(acc, offset, Flags{});
+  constexpr unsigned Size = sizeof(T) * N;
+  constexpr unsigned Align = Flags::template alignment<T1>;
+
+  simd<T, N> Tmp;
+  if constexpr (Align >= OperandSize::DWORD && Size % OperandSize::OWORD == 0 &&
```

Comment:
Block load requires offset to be at least dword aligned, so it should be OperandSize::DWORD. block_load() then will check what alignment is and will use either aligned load if alignment is >= OperandSize::OWORD or unaligned otherwise.